src/workflow/test/RunPreProcessMonthlyReports_GTest.cpp

```diff
@@ -0,0 +1,138 @@
+    boost::optional<openstudio::InstallLocationType> result;
+    boost::smatch matches;
+    if (boost::regex_search(name, matches, meterRegex)) {
+      if (matches[4].matched) {
+        result = openstudio::InstallLocationType(std::string(matches[4].first, matches[4].second));
+      }
+    }
+
+    return result;
+  };
+
+  for (auto meter : c_monthlyReports) {
+    auto idfObject_ = openstudio::IdfObject::load(std::string{meter});
+
+    ASSERT_TRUE(idfObject_);
+    auto idfObject = std::move(*idfObject_);
+    auto name = idfObject.nameString();
+    SCOPED_TRACE("Processing " + name + "\n========================================");
+    const auto n = name.find(" - ");
+    auto tableFuelType = name.substr(n + 3);
+
+    EXPECT_NE(n, std::string::npos);
+    auto prefix = name.substr(0, n);
+    EXPECT_EQ(expectedPrefix, prefix);
+
+    auto demandN = tableFuelType.find(peakDemandSuffix);
+    bool isDemand = false;
+    if (demandN != std::string::npos) {
+      isDemand = true;
+      ++n_demand;
+      tableFuelType = tableFuelType.substr(0, tableFuelType.size() - peakDemandSuffix.size());
+    } else {
+      ++n_regular;
+    }
+    try {
+      auto tableEUFT = openstudio::EndUseFuelType(tableFuelType);
+      EXPECT_EQ(tableFuelType, tableEUFT.valueDescription());
+    } catch (...) {
+      EXPECT_TRUE(false) << tableFuelType << " is not a valid EndUseFuelType";
+    }
+
+    EXPECT_EQ(2, idfObject.getInt(1).get());
+    const std::string aggregrationType = isDemand ? "ValueWhenMaximumOrMinimum" : "SumOrAverage";
+    bool is_first = true;
+    for (const auto& eg : idfObject.extensibleGroups()) {
+      const auto meterName = eg.getString(0).get();
+      SCOPED_TRACE("Processing extensible group " + meterName);
+
+      const auto eut = endUseType(meterName);
+      const auto ft = fuelType(meterName);
+      const auto loc = installLocationType(meterName);
+
+      const auto reconstructedMeterName = openstudio::model::OutputMeter::getName(boost::none, eut, ft, loc, boost::none);
+
+      // This is the definitive test
+      EXPECT_EQ(reconstructedMeterName, meterName);
+
+      if (isDemand && is_first) {
+        // Electricity:Facility : Maximum
+        EXPECT_FALSE(eut) << "EndUseType should not be present";
+        EXPECT_TRUE(ft) << "FuelType should be present";
+        ASSERT_TRUE(loc) << "InstallLocationType should be present";
+        EXPECT_EQ("Facility", loc->valueName());
+
+        EXPECT_EQ("Maximum", eg.getString(1).get());
+
+        is_first = false;
+        continue;
+      }
+
+      EXPECT_TRUE(eut) << "EndUseType should be present";
+      EXPECT_TRUE(ft) << "FuelType should be present";
+      EXPECT_FALSE(loc) << "InstallLocationType should not be present";
+
+      EXPECT_EQ(aggregrationType, eg.getString(1).get());
+    }
+  }
+
+  EXPECT_EQ(n_regular, n_demand);
+  EXPECT_EQ(4, n_regular);
```

Comment:
Annoying to write in C++, but this test will ensure that this doesn't happen again.